packages/astro/src/core/render/dev/index.ts

```diff
@@ -69,10 +75,19 @@ export async function render(renderers: Renderer[], mod: ComponentInstance, ssrO
 			children: '',
 		});
 		scripts.add({
-			props: { type: 'module', src: new URL('../../../runtime/client/hmr.js', import.meta.url).pathname },
+			props: { type: 'module', src: '/@id/astro/client/hmr.js' },
```

Comment:
Wish I knew about this magic `/@id` sooner.

Reply:
Vite always keeps you guessing :)

Reply:
THIS WORKS WHAT THIS IS GAME CHANGING